### PR TITLE
Issue 547: Add support for init containers in controller and segment store pods

### LINKED ISSUE #547: Add support for init containers in controller and segment store pods
### Description

Provide support to add init containers to controller and segment store pods

### Importance
must-have

### Location

pkg/controller/pravega/pravega_segmentstore.go
pkg/controller/pravega/pravega_controller.go

### Suggestions for an improvement
Add spec fields for init containers

### PR DESCRIPTION
**Pull request by anishakj**
### Change log description

Provide support for adding init containers to pravega pods
### Purpose of the change

 Fixes #547

### What the code does

Added a new spec parameter for init containers,  and if user  has provided init containers, add it to the podspec
### How to verify it

Verified that init containers are coming up in controller and segment store pods and mentioned command is getting executed correctly

### CODE REVIEW

File: deploy/crds/pravega.pravega.io_pravegaclusters_crd.yaml [hunk 1]
```diff
@@ -253,6 +253,1093 @@ spec:
                       access. Options are "LoadBalancer" and "NodePort". By default,
                       if external access is enabled, it will use "LoadBalancer"
                     type: string
+                  ontrollerInitContainers:
```
Comment (SrishT) [inline deploy/crds/pravega.pravega.io_pravegaclusters_crd.yaml:256]: ontrollerInitContainers needs to be corrected

File: pkg/controller/pravega/pravega_segmentstore_test.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/pravega/pravega-operator/pkg/controller/pravega"
 
 	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (SrishT) [inline pkg/controller/pravega/pravega_segmentstore_test.go:22]: v1 has already been imported by the name corev1 in the previous line

File: pkg/controller/pravega/pravega_controller_test.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/pravega/pravega-operator/pkg/controller/pravega"
 
 	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (SrishT) [inline pkg/controller/pravega/pravega_controller_test.go:22]: v1 has already been imported by the name corev1 in the previous line

**Review by SrishT — CHANGES_REQUESTED** [review]

File: deploy/crds/pravega.pravega.io_pravegaclusters_crd.yaml (hunk 1 above)
Comment (anishakj) [reply deploy/crds/pravega.pravega.io_pravegaclusters_crd.yaml]: fixed

File: pkg/controller/pravega/pravega_segmentstore_test.go (hunk 2 above)
Comment (anishakj) [reply pkg/controller/pravega/pravega_segmentstore_test.go]: fixed

File: pkg/controller/pravega/pravega_controller_test.go (hunk 3 above)
Comment (anishakj) [reply pkg/controller/pravega/pravega_controller_test.go]: fixed

**Review by SrishT — APPROVED** [review]
LGTM

**Pull request merged**